provider/aws/aws_discover.go

```diff
@@ -80,7 +88,8 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 
 	l.Printf("[DEBUG] discover-aws: Creating session...")
 	svc := ec2.New(session.New(), &aws.Config{
-		Region: &region,
+		Region:   &region,
+		Endpoint: endpoint,
```

Comment:
What happens when the endpoint URL is blank? Should we be doing a conditional and setting it against the `aws.Confg{}` selectively? That seems safer as the client may handle an empty string differently than we expect, and we don't want `endpoint_url` to be required.

Reply:
aws.Config.Endpoint is a pointer to a string so the default value is nil.
Here: https://github.com/hashicorp/go-discover/pull/108/files#diff-a1d11610f4fd8510669547475b8c6270R73 I'm checking if endpoint_url is blank and in that case endpoint value is just left to nil, so when is set on the aws.Config, is just the default value for that property.

Reply:
Thanks for the explanation, that makes sense to me!

We should add a note that this has a default, as some of the other options to in other providers. Maybe we can figure out how the AWS client derives the default and briefly note that?

Reply:
Hi, I made a comment about the default value of that argument.